### PR TITLE
Log a final, machine readable status

### LINKED ISSUE #213: Add final STATUS line to 'output files' i.e. /var/lib/auter/last-{phase}-output-default
I've a script which I run across multiple servers to determine the success/failure status of each phase of auter for reporting purposes.

/var/lib/auter/last-{phase}-output-default  is generated for each phase of auter, and overwritten with the output from the latest run each time.

We wish to be able to determine, by examining this file, whether or not this specific phase of the auter run has succeeded or failed at the latest attempt. 
---

The following approaches are currently possible but have downsides:
A - grep for the auter logs in /var/log/messages. Look for 'auter --{phase} ' and then look for any 'ERROR' which follows.
  - If there have been re-runs of e.g. the auter --prep phase after fixing an issue it's tricky to determine whether an ERROR if for the latest or previous iterations
 - If a prep and an apply happen shortly after each other - again disambiguating whether an ERROR message belongs to which phase is tricky
 - It can be done with sed/awk/etc, but is open to error and bad assumptions

B - find the datestamp at the top of the output file, look for ERRORs  in the system log at that time.
 - The timestamp in the file records the start of the auter run, the timestamp of the ERROR in the system log records some time later (maybe only a few seconds or minutes) - things could be done to e.g. only look for ERRORs in the same hour or minute or day, but this leads to unreliability should the auter run cross a time boundary. We could try to get clever with converting log timestamps to seconds since epoch, and then do something with that, but it's still not going to be that reliable.

C - grep -i ERROR in the 'output file' for that phase
- It has a timestamp at the top which we can use
- We can look for variations of 'Error|error|ERROR' etc in the log and report a fail if it finds it.
- False negatives if there are any rpms being installed which contain the word 'error' in them.
- We could look for specific error strings such as 'Error Downloading Packages' but this might miss other error conditions.

---

Instead could you add a status line to the end of the /var/lib/auter/last-{phase}-output-default files with an easily parse-able status

### PR DESCRIPTION
**Pull request by rhodesn**
Resolves #213 

Log to /var/lib/auter/last-{prep,apply}-output-X with either:
```
STATUS:SUCCESS:Message
-or-
STATUS:FAILED:Message
```

### CODE REVIEW

File: auter.yumdnfModule [hunk 1]
```diff
@@ -164,9 +169,13 @@ function apply_updates() {
...
     rotate_file "${DATADIR}/last-apply-output-${CONFIGSET}"
-    APPLYOUTPUT+=$("${PACKAGE_MANAGER}" "${UPDATEACTION}" -y "${PACKAGEMANAGEROPTIONS[@]}" "${RPMS[@]}" 2>&1)
+    if APPLYOUTPUT+=$("${PACKAGE_MANAGER}" "${UPDATEACTION}" -y "${PACKAGEMANAGEROPTIONS[@]}" "${RPMS[@]}" 2>&1); then
+      APPLYOUTPUT+="\\nSTATUS:SUCCESS:Package updates applied"
+    else
+      APPLYOUTPUT+="\\nSTATUS:FAILED:Package updates failed"
+    fi
     default_signal_handling
-    echo -e "$APPLYOUTPUT" &>"${DATADIR}/last-apply-output-${CONFIGSET}"
+    echo -e "$APPLYOUTPUT" > "${DATADIR}/last-apply-output-${CONFIGSET}"
```
Comment (mark-hyde) [inline auter.yumdnfModule:179]: Wondering why the change from `&>` to `>` ?

**Review by mark-hyde — APPROVED** [review]
Looks good to me.
Wondering why you went from `&>` to `>` in one of the lines?

File: auter.yumdnfModule (hunk 1 above)
Comment (rhodesn) [reply auter.yumdnfModule:179]: I couldn't work out why we would need stderr from an echo. Have I missed something?

Comment (mark-hyde) [reply auter.yumdnfModule:179]: Good point!
I was thinking it was from the output of the commands themselves! 
Approved then!! :)

**Review by mark-hyde — DISMISSED** [review]

File: auter.yumdnfModule (hunk 1 above)
Comment (rhodesn) [reply auter.yumdnfModule:179]: Awesome, thanks for pulling it up! I'll look at getting it merged and into a release.
